### ISSUE TITLE
TIP: RuntimeError: unreachable

### ISSUE THREAD

**Issue by fw-brice** [v8-riscv/v8#289]
**Describe the bug**
This issue shows up when using the `rebase-to-tip` branch (#273).

**To Reproduce**
1. Checkout `rebase-to-tip`
2. Follow standard build procedure
3. Run the test:
```
out/riscv64.sim/d8 --test /v8/test/wasm-spec-tests/tests/block.js --random-seed=-1445372902 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
```

**Expected behavior**
I expected this test to pass.

**Errors/Logs**
```
Warning: unknown flag --enable-slow-asserts.
Try --help for options
wasm-function[1]:0x76: RuntimeError: unreachable
RuntimeError: unreachable
    at run (<anonymous>:wasm-function[1]:0x76)
    at call (/v8/test/wasm-spec-tests/tests/block.js:51:32)
    at /v8/test/wasm-spec-tests/tests/block.js:270:11
    at run (/v8/test/wasm-spec-tests/tests/block.js:64:3)
    at /v8/test/wasm-spec-tests/tests/block.js:270:1
```

**Possibly related failures:**
* wasm-spec-tests/if
* wasm-spec-tests/loop

**Comment by luyahan** [comment]
The error  randomly appears !!
```
yahan@2b971f1ad76e:~/v8-riscv64 $ ./out/riscv64.sim/d8 --test ./test/wasm-spec-tests/tests/block.js --random-seed=-1445372902 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
Warning: unknown flag --enable-slow-asserts.
Try --help for options
yahan@2b971f1ad76e:~/v8-riscv64 $ ./out/riscv64.sim/d8 --test ./test/wasm-spec-tests/tests/block.js --random-seed=-1445372902 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
Warning: unknown flag --enable-slow-asserts.
Try --help for options
wasm-function[1]:0x76: RuntimeError: unreachable
RuntimeError: unreachable
    at run (<anonymous>:wasm-function[1]:0x76)
    at call (./test/wasm-spec-tests/tests/block.js:51:32)
    at ./test/wasm-spec-tests/tests/block.js:270:11
    at run (./test/wasm-spec-tests/tests/block.js:64:3)
    at ./test/wasm-spec-tests/tests/block.js:270:1

yahan@2b971f1ad76e:~/v8-riscv64 $ ./out/riscv64.sim/d8 --test ./test/wasm-spec-tests/tests/block.js --random-seed=-1445372902 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
Warning: unknown flag --enable-slow-asserts.
Try --help for options
wasm-function[1]:0x76: RuntimeError: unreachable
RuntimeError: unreachable
    at run (<anonymous>:wasm-function[1]:0x76)
    at call (./test/wasm-spec-tests/tests/block.js:51:32)
    at ./test/wasm-spec-tests/tests/block.js:270:11
    at run (./test/wasm-spec-tests/tests/block.js:64:3)
    at ./test/wasm-spec-tests/tests/block.js:270:1

yahan@2b971f1ad76e:~/v8-riscv64 $ ./out/riscv64.sim/d8 --test ./test/wasm-spec-tests/tests/block.js --random-seed=-1445372902 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
Warning: unknown flag --enable-slow-asserts.
Try --help for options
wasm-function[1]:0x76: RuntimeError: unreachable
RuntimeError: unreachable
    at run (<anonymous>:wasm-function[1]:0x76)
    at call (./test/wasm-spec-tests/tests/block.js:51:32)
    at ./test/wasm-spec-tests/tests/block.js:270:11
    at run (./test/wasm-spec-tests/tests/block.js:64:3)
    at ./test/wasm-spec-tests/tests/block.js:270:1

yahan@2b971f1ad76e:~/v8-riscv64 $
yahan@2b971f1ad76e:~/v8-riscv64 $ ./out/riscv64.sim/d8 --test ./test/wasm-spec-tests/tests/block.js --random-seed=-1445372902 --nohard-abort --enable-slow-asserts --verify-heap --testing-d8-test-runner
Warning: unknown flag --enable-slow-asserts.
Try --help for options
yahan@2b971f1ad76e:~/v8-riscv64 $
```

**Comment by fw-brice** [comment]
After the fix to #287, I haven't seen this one fail again. We may be able to close this.
